libs/partners/ibm/langchain_ibm/llms.py

```diff
@@ -0,0 +1,396 @@
+                    " as a named parameter."
+                )
+            elif values["token"] or "WATSONX_TOKEN" in os.environ:
+                values["token"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "token", "WATSONX_TOKEN")
+                )
+            elif values["password"] or "WATSONX_PASSWORD" in os.environ:
+                values["password"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "password", "WATSONX_PASSWORD")
+                )
+                values["username"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "username", "WATSONX_USERNAME")
+                )
+            elif values["apikey"] or "WATSONX_APIKEY" in os.environ:
+                values["apikey"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "apikey", "WATSONX_APIKEY")
+                )
+                values["username"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "username", "WATSONX_USERNAME")
+                )
+            if not values["instance_id"] or "WATSONX_INSTANCE_ID" not in os.environ:
+                values["instance_id"] = convert_to_secret_str(
+                    get_from_dict_or_env(values, "instance_id", "WATSONX_INSTANCE_ID")
+                )
+
+        try:
+            from ibm_watsonx_ai.foundation_models import ModelInference  # type: ignore
+
+            credentials = {
+                "url": values["url"].get_secret_value() if values["url"] else None,
+                "apikey": values["apikey"].get_secret_value()
+                if values["apikey"]
+                else None,
+                "token": values["token"].get_secret_value()
+                if values["token"]
+                else None,
+                "password": values["password"].get_secret_value()
+                if values["password"]
+                else None,
+                "username": values["username"].get_secret_value()
+                if values["username"]
+                else None,
+                "instance_id": values["instance_id"].get_secret_value()
+                if values["instance_id"]
+                else None,
+                "version": values["version"].get_secret_value()
+                if values["version"]
+                else None,
+            }
+            credentials_without_none_value = {
+                key: value for key, value in credentials.items() if value is not None
+            }
+
+            watsonx_model = ModelInference(
+                model_id=values["model_id"],
+                deployment_id=values["deployment_id"],
+                credentials=credentials_without_none_value,
+                params=values["params"],
+                project_id=values["project_id"],
+                space_id=values["space_id"],
+                verify=values["verify"],
+            )
+            values["watsonx_model"] = watsonx_model
+
+        except ImportError:
```

Comment:
can remove - this should never happen because it's a dep!

Reply:
Removed